gradio/blocks.py

```diff
@@ -768,34 +821,61 @@ async def process_api(
         fn_index: int,
         inputs: List[Any],
         username: str = None,
-        state: Dict[int, Any] | None = None,
+        state: Dict[int, Any] | List[Dict[int, Any]] | None = None,
         iterators: Dict[int, Any] | None = None,
     ) -> Dict[str, Any]:
         """
         Processes API calls from the frontend. First preprocesses the data,
         then runs the relevant function, then postprocesses the output.
         Parameters:
-            data: data recieved from the frontend
-            inputs: the list of raw inputs to pass to the function
+            fn_index: Index of function to run.
+            inputs: input data received from the frontend
             username: name of user if authentication is set up (not used)
             state: data stored from stateful components for session (key is input block id)
             iterators: the in-progress iterators for each generator function (key is function index)
         Returns: None
         """
         block_fn = self.fns[fn_index]
+        batch = self.dependencies[fn_index]["batch"]
+
+        if batch:
+            max_batch_size = self.dependencies[fn_index]["max_batch_size"]
+            batch_sizes = [len(inp) for inp in inputs]
+            batch_size = batch_sizes[0]
+            if inspect.isasyncgenfunction(block_fn.fn) or inspect.isgeneratorfunction(
+                block_fn.fn
+            ):
+                raise ValueError("Gradio does not support generators in batch mode.")
+            if not all(x == batch_size for x in batch_sizes):
+                raise ValueError(
+                    f"All inputs to a batch function must have the same length but instead have sizes: {batch_sizes}."
+                )
+            if batch_size > max_batch_size:
+                raise ValueError(
```

Comment:
We should raise an error when you try to batch a function and the queue is not enabled.

Reply:
Added error and tests to confirm the error is called